sys/kern/thread.c

```diff
@@ -50,6 +51,7 @@ static void thread_init(thread_t *td, prio_t prio) {
   td->td_lock = MTX_INITIALIZER(0);
   cv_init(&td->td_waitcv, "thread waiters");
   LIST_INIT(&td->td_contested);
+  bzero(&td->td_slpcallout, sizeof(callout_t));
```

Comment:
I think we should also drain the thread's callout before freeing it in `thread_delete`, otherwise the callout function (e.g. `sq_timeout`) may refer to freed memory.

Reply:
Right! I've applied corresponding change.